File: python/Vehicle.py
#Saya Amelia Zalfa Julianti dengan NIM 2203999 mengerjakan Latihan 4 mata kuliah DPBO untuk keberkahanNya maka saya tidak melakukan kecurangan sesuai dengan apa yang telah dispesifikasikan. Aamiin.
class Vehicle:
    __plat = ""
    __merk = ""
    __tahunProduksi = ""
    __warna = ""
    
    def __init__(self, plat = "", merk = "", tahunProduksi = "", warna = ""):
        self.__plat = plat
        self.__merk = merk
        self.__tahunProduksi = tahunProduksi
        self.__warna = warna
        
    #Getter and Setter
    
    #Get plat
    
    def get_plat(self):
        return self.__plat
    
    #set plat
    def set_plat(self, plat):
        self.__plat = plat
    
    #Get merk
    
    def get_merk(self):
        return self.__merk
    
    #set merk
    def set_merk(self, merk):
        self.__merk = merk
        
    #Get tahunProduksi
    
    def get_tahunProduksi(self):
        return self.__tahunProduksi
    
    #set tahunProduksi
    def set_tahunProduksi(self, tahunProduksi):
        self.__tahunProduksi = tahunProduksi
        
    #Get warna
    
    def get_warna(self):
        return self.__warna
    
    #set warna
    def set_warna(self, warna):
        self.__warna = warna
    

File: cpp/Vehicle.cpp
#pragma once

//memasukkan library yang digunakan

#include <iostream>
#include <string>

//deklarasi namespace
#
using namespace std;

class Vehicle {
	private:
	string plat;
	string merk;
	string tahunProduksi;
	string warna;

	public:

	Vehicle(){

	}

	Vehicle(string plat) {
		this->plat = plat;
		merk = "";
		tahunProduksi = "";
		warna = "";
	}

	Vehicle(string plat, string merk) {
		this->plat = plat;
		this->merk = merk;
		tahunProduksi = "";
		warna = "";
	}

	Vehicle(string plat, string merk, string tahunProduksi){
		this->plat = plat;
		this->merk = merk;
		tahunProduksi = tahunProduksi;
		warna = "";
	}

	Vehicle(string plat, string merk, string tahunProduksi, string warna){
		this->plat = plat;
		this->merk = merk;
		this->tahunProduksi = tahunProduksi;
		warna = warna;
	}

	// Getter and Setter

	// Get plat
	string getPlat() {
		return plat;
	}

	// Set plat
	void set_plat(string plat) {
		this->plat = plat;
	}

	// Get merk
	string get_merk() {
		return merk;
	}

	// Set merk
	void set_merk(string merk) {
	 	this->merk = merk;
	}

	// Get tahunProduksi
	string get_tahunProduksi() {
		return tahunProduksi;
	}

	// Set tahunProduksi
	void set_tahunProduksi(string tahunProduksi) {
		this->tahunProduksi = tahunProduksi;
	}

	// Get warna
	string get_warna() {
		return warna;
	}

	// Set warna
	void set_warna(string warna) {
		warna = warna;
	}

	~Vehicle(){

	}
};
